sql/sqlwire/wire.proto

```diff
@@ -102,6 +100,16 @@ message Request {
   repeated Datum params = 3 [(gogoproto.nullable) = false];
 }
 
+// ResponseWithError is what we send on the wire. It includes the error so that
+// Response doesn't need to, simplifying error handling that involves handling
+// these objects.
+message ResponseWithError {
```

Comment:
I've acquiesced to your demands, PTAL

On Fri, Jul 10, 2015 at 4:01 PM, Peter Mattis <notifications@github.com>
wrote:

> In sql/sqlwire/wire.proto
> <https://github.com/cockroachdb/cockroach/pull/1661#discussion_r34390821>:
>
> > @@ -102,6 +100,16 @@ message Request {
> >    repeated Datum params = 3 [(gogoproto.nullable) = false];
> >  }
> >
> > +// ResponseWithError is what we send on the wire. It includes the error so that
> > +// Response doesn't need to, simplifying error handling that involves handling
> > +// these objects.
> > +message ResponseWithError {
>
> I too like using the compiler to make changes like this. You could have
> accomplished the same by renaming SetGoError to something like SetError
> or moving it to the sqlserver package now that it only has a single use.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1661/files#r34390821>.
>